### PR TITLE
Ap 5582 - prep for Magento Marketplace

### PR DESCRIPTION
**Pull request by Pro777**

### CODE REVIEW

File: app/code/Apruve/Payment/Controller/Data/Index.php [hunk 1]
```diff
@@ -57,51 +58,52 @@ public function execute() {
...
-                    unset ($order['line_items'][$k]);
+                    unset($order['line_items'][$k]);
                 }
             }
         }
 
         $this->order = $order;
-        $data = array(
+        $data = [
             'order' => json_encode($this->order),
```
Comment (donnen) [inline app/code/Apruve/Payment/Controller/Data/Index.php]: Is this going to be double-encoded now that the return value is also being encoded?

File: app/code/Apruve/Payment/Controller/updateOrderStatus/Index.php [hunk 2]
```diff
@@ -22,138 +22,174 @@ public function __construct(
...
-                
-                $this->_acceptOrder($data);
-            break;
-            
             case 'invoice.closed':
                 $this->_invoiceClosed($data);
-            break;
+                break;
+
+            case 'order.canceled':
```
Comment (donnen) [inline app/code/Apruve/Payment/Controller/updateOrderStatus/Index.php]: Is this a duplicate of line 60?

File: app/code/Apruve/Payment/Controller/updateOrderStatus/Index.php [hunk 3]
```diff
@@ -22,138 +22,174 @@ public function __construct(
...
-        return file_get_contents('php://input');
-    }
-    
-    protected function _getData() {
-        $data = trim($this->_getRawData());
-        return json_decode($data);
+
+    protected function _createInvoice()
+    {
+//        if ($orderId) {
```
Comment (donnen) [inline app/code/Apruve/Payment/Controller/updateOrderStatus/Index.php]: Is this supposed to be commented out?

File: app/code/Apruve/Payment/Model/PaymentMethod.php [hunk 4]
```diff
@@ -7,248 +7,300 @@ class PaymentMethod extends \Magento\Payment\Model\Method\AbstractMethod
...
+        $data                        = [];
+        $data['merchant_id']         = $this->_getMerchantKey();
+        $data['merchant_order_id']   = $order->getIncrementId();
+        $data['shopper_id']          = $buyer->id;
+        $data['amount_cents']        = ( $order->getPayment()->getData('amount_ordered') ) * 100;
+        $data['currency']            = $order->getData('base_currency_code');
+        $data['shipping_cents']      = $order->getData('shipping_amount') * 100;
+        $data['tax_cents']           = $order->getData('tax_amount') * 100;
+        $data['merchant_notes']      = '';
+        $data['merchant_invoice_id'] = 'blah!';
```
Comment (tcochran317) [inline app/code/Apruve/Payment/Model/PaymentMethod.php]: I think we'd prefer this stay unset or blank if there's no Magento internal invoice ID available here.

**Pull request merged**